tmail-backend/combined-identity/src/test/java/com/linagora/tmail/combined/identity/CombinedUsersRepositoryContract.java

```diff
@@ -1,58 +1,513 @@
+            //Given
+            testee().addUser(testSystem.userAlreadyInLDAP, "password");
+            //When
+            User actual = testee().getUserByName(testSystem.userAlreadyInLDAPCaseVariation);
+            //Then
+            assertThat(actual).isNotNull();
+            assertThat(actual.getUserName()).isEqualTo(testSystem.userAlreadyInLDAP);
+        }
+
+        @Test
+        default void testShouldReturnTrueWhenAUserHasACorrectPassword(CombinedTestSystem testSystem) throws UsersRepositoryException {
+            boolean actual = testee().test(testSystem.userAlreadyInLDAP, "secret");
+            assertThat(actual).isTrue();
+        }
+
+        @Test
+        default void testShouldReturnFalseWhenAUserHasAnIncorrectPassword(CombinedTestSystem testSystem) throws UsersRepositoryException {
+            boolean actual = testee().test(testSystem.userAlreadyInLDAP, "password2");
+            assertThat(actual).isFalse();
+        }
+
+        @Test
+        default void testShouldReturnFalseWhenAUserHasAnIncorrectCasePassword(CombinedTestSystem testSystem) throws UsersRepositoryException {
+            boolean actual = testee().test(testSystem.userAlreadyInLDAP, "Secret");
+
+            assertThat(actual).isFalse();
+        }
+
+        @Test
+        default void testShouldReturnFalseWhenAUserIsNotInRepository(CombinedTestSystem testSystem) throws UsersRepositoryException {
+            boolean actual = testee().test(testSystem.userNotAlreadyInLDAP, "secret");
+            assertThat(actual).isFalse();
+        }
+
+        @Test
+        default void testShouldReturnTrueWhenAUserHasAnIncorrectCaseName(CombinedTestSystem testSystem) throws UsersRepositoryException {
+            boolean actual = testee().test(testSystem.userAlreadyInLDAPCaseVariation, "secret");
+
+            assertThat(actual).isTrue();
+        }
+
+        @Ignore("This feature is not supported in Combined Repository")
+        @Test
+        default void testShouldReturnFalseWhenAUserIsRemovedFromRepository(CombinedTestSystem testSystem) throws UsersRepositoryException {
+        }
+
+        @Test
+        default void removeUserShouldRemoveAUserWhenPresentInRepository(CombinedTestSystem testSystem) throws UsersRepositoryException {
+            cassandraUsersDAO().addUser(testSystem.userNotAlreadyInLDAP, "password");
+            testee().removeUser(testSystem.userNotAlreadyInLDAP);
+
+            assertThat(testee().contains(testSystem.userNotAlreadyInLDAP)).isFalse();
+        }
+
+        @Ignore("This feature is not supported in Combined Repository")
+        @Test
+        default void updateUserShouldAllowToAuthenticateWithNewPassword(CombinedTestSystem testSystem) throws UsersRepositoryException {
+        }
+
+        @Ignore("This feature is not supported in Combined Repository")
+        @Test
+        default void updateUserShouldNotAllowToAuthenticateWithOldPassword(CombinedTestSystem testSystem) throws UsersRepositoryException {
```

Comment:
Then we can remove all these `This feature is not supported in Combined Repository` tests